src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py

```diff
@@ -738,6 +738,60 @@ def has_pending_clones(self, snapname):
                 return False
             raise
 
+    def add_orphan_clones(self, track_id, snapname):
+        self.metadata_mgr.add_section('orphan clones')
+        self.metadata_mgr.update_section('orphan clones', track_id, snapname)
+        self.metadata_mgr.flush()
+
+    def get_pending_clones(self, snapname):
+        pending_clones = {}  # type: Dict[str, Any]
+        pending_track_id_list = []
+        pending_clone_list = []
+        index_path = ""
+
+        try:
+            if self.has_pending_clones(snapname):
+                pending_track_id_list = self.metadata_mgr.list_all_keys_with_specified_values_from_section('clone snaps', snapname)
+        except MetadataMgrException as me:
+            if me.errno != -errno.ENOENT:
+                raise
+
+        try:
+            with open_clone_index(self.fs, self.vol_spec) as index:
+                index_path = index.path.decode('utf-8')
+        except IndexException as e:
+            log.warning("failed to open clone index: {0}".format(e))
+            raise VolumeException(-errno.EINVAL, "failed to open clone index")
+
+        for track_id in pending_track_id_list:
+            try:
+                link_path = self.fs.readlink(os.path.join(index_path, track_id), 4096)
+                pending_clone_list.append(os.path.basename(link_path).decode('utf-8'))
+            except cephfs.Error as e:
+                if e.errno != errno.ENOENT:
+                    raise
+                else:
+                    try:
+                        # If clone is completed between 'list_all_keys_with_specified_values_from_section'
+                        # and readlink(track_id_path) call then readlink will fail with error ENOENT (2)
+                        # Hence we double check whether track_id is exist in .meta file or not.
+                        value = self.metadata_mgr.get_option('clone snaps', track_id)
+                        # Edge case scenario.
+                        # If track_id for clone exist but path /volumes/_index/clone/{track_id} not found
+                        # then add track_id in the orphan clones section.
+                        self.add_orphan_clones(track_id, value)
```

Comment:
Please add a test case for the orphaned_clone usecase. You might have to tweak manually to reproduce this.